### ISSUE TITLE
Not Scrolling on Pan gesture - issue when adding in TAB Bar controller ,

### ISSUE THREAD

**Issue by prashantAlgante** [OfTheWolf/UBottomSheet#35]
Hello,

When i am loading screen first time then the Bottomsheet stuck at the initial position not expanding to higher position. When i am going to some other screen in navigation and coming back to initial screen then Bottom sheet starts expanding. This is repeating ever time on first time load. Can you help me out.

**Comment by prashantAlgante** [comment]
For you information 
I also tried putting this library in TabbarViewController class and hiding default tabbar. This loads perfectly but Does not scroll to higher position at all. Please advise something , what needs to be done in second case

Below is the code attached from your example 


```
class CustomTabBarVC: UITabBarController {
    var sheetCoordinator: UBottomSheetCoordinator!
    
    var sheetVC: DraggableItem!
    var useNavController = false
    var dataSource: UBottomSheetCoordinatorDataSource?
    
    override func viewDidLoad() {
        super.viewDidLoad()

        self.tabBar.isHidden = true
        self.sheetVC = ListViewController()
        self.dataSource = MyDataSource()
    }
        
    override func viewWillLayoutSubviews() {
        super.viewWillLayoutSubviews()
        
        guard sheetCoordinator == nil else {return}
        print("Called from subviews")
        sheetCoordinator = UBottomSheetCoordinator(parent: self)
        if dataSource != nil{
            sheetCoordinator.dataSource = dataSource!
        }

        let vc: UIViewController
        if useNavController{
            vc = UINavigationController(rootViewController: sheetVC)
            sheetVC.title = "Navigation Bar Title"
        }else{
            vc = sheetVC
        }
        sheetVC.sheetCoordinator = sheetCoordinator
        sheetCoordinator.addSheet(vc, to: self, didContainerCreate: { container in
            let f = self.view.frame
            let rect = CGRect(x: f.minX, y: f.minY, width: f.width, height: f.height)
            container.roundCorners(corners: [.topLeft, .topRight], radius: 10, rect: rect)
        })
    }

}

```

**Comment by OfTheWolf** [comment]
@prashantAlgante I will try to look at it this week asap.

**Comment by prashantAlgante** [comment]
Hello @OfTheWolf  Can you please also look for #36 . That will also be much simpler for you to do. Thanks

**Comment by GerdC** [comment]
**Edit: my fault. See my comment below** 

I think I have the same problem.

A MapViewController inside a `UITabBarController`
The MapViewController controller presents the Sheet controller.

This is an actual screenshot using UBottomSheet:

![Simulator Screen Shot - iPhone SE (2nd generation) - 2020-08-07 at 19 42 33](https://user-images.githubusercontent.com/7478802/89673463-b4770c80-d8e6-11ea-9d28-a961ecb3b91a.png)

The SheetViewController has no `UIScrollView` but implements Draggable.

I use this code from the MapViewController:
        
        let dataSource = StackDataSourceProxy(viewController: transitPlanViewController)
        self.stackDataSourceProxy = dataSource // strong reference
        self.sheetCoordinator.dataSource = dataSource // weak reference

        sheetCoordinator.addSheet(transitPlanViewController, to: self, didContainerCreate: { container in
            let f = self.view.frame
            let rect = CGRect(x: f.minX, y: f.minY, width: f.width, height: f.height)
            container.roundCorners(corners: [.topLeft, .topRight], radius: 10, rect: rect)
        })
        sheetCoordinator.setCornerRadius(10)
        
Expected behaviour: scrolling on pan gesture.
Actual behaviour: not scrolling on pan gesture.

(availableHeight does not respect the tabBar height but that's easy to work around)

**Comment by OfTheWolf** [comment]
@prashantAlgante @GerdC 
Interesting bug. It seems TabBarController causes child vc's viewWillAppear is not called. So as a workaround put all startTracking logics in viewWillLayoutSubviews.

   ``` 
//***ListViewController.swift***

    var isTracking = false

    override func viewWillLayoutSubviews() {
        super.viewWillLayoutSubviews()
        
        if !isTracking{
            isTracking = true
            sheetCoordinator?.startTracking(item: self)
        }
    }
```

**Comment by OfTheWolf** [comment]
> (availableHeight does not respect the tabBar height but that's easy to work around)

This should be the problem.
@GerdC    

```
    public var availableHeight: CGFloat{
        if #available(iOS 11.0, *) {
            return parent.view.frame.inset(by: parent.view.safeAreaInsets).height
        } else {
            return parent.view.frame.height
        }
    }
```

**Comment by GerdC** [comment]
> @prashantAlgante @GerdC
> Interesting bug. It seems TabBarController causes child vc's viewWillAppear is not called. So as a workaround put all startTracking logics in viewWillLayoutSubviews.
> 
> ```
> //***ListViewController.swift***
> 
>  var isTracking = false
> 
>  override func viewWillLayoutSubviews() {
>      super.viewWillLayoutSubviews()
>      
>      if !isTracking{
>          isTracking = true
>          sheetCoordinator?.startTracking(item: self)
>      }
>  }
> ```


it was my fault, but your answer was a great help to find it.
I did not set the sheetCoordinator in the presented sheet, only in the presenting controller.

Will never happen again:

    public override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        if let sheetCoordinator = self.sheetCoordinator {
            sheetCoordinator.startTracking(item: self)
        } else {
            assertionFailure()
        }
    }

I don't think any more that there is a problem using UBottomSheet in a TabBarController.
You could add an assertion to your example code as I did to make it easy to find this type of mistake.

Thanks
